readthedocs/doc_builder/environments.py

```diff
@@ -373,6 +353,34 @@ class LocalEnvironment(BaseEnvironment):
     # TODO: BuildCommand name doesn't make sense here, should be just Command
     command_class = BuildCommand
 
+    def post_run_command(self):
+        command = self.commands[-1]
+        if self.record_as_success:
+            command.exit_code = 0
+        if self.record:
+            command.save()
+
+    def run(self, *cmd, **kwargs):
+        self.record = kwargs.pop('record', False)
+        self.record_as_success = kwargs.pop('record_as_success', False)
+        if not self.record:
+            kwargs['warn_only'] = True
+        if self.record_as_success:
+            self.record = True
+            kwargs['warn_only'] = True
+        return super(LocalEnvironment, self).run(*cmd, **kwargs)
+
+    # record, force_success, warn_only
```

Comment:
I see that there is some _magic_ with the parameters on the `run_command_class`, change according to others parameters. So I was thinking to use different parameters: record, force_success, warn_only that are independent from each other and play nice together.